cmake/modules/snippets.cmake

```diff
@@ -53,11 +53,20 @@ function(zephyr_process_snippets)
     file(REMOVE ${snippets_generated})
   else()
     string(REPLACE " " ";" SNIPPET_AS_LIST "${SNIPPET}")
+    if(Sysbuild_DIR)
+      string(REPLACE " " ";" SYSBUILD_SNIPPET_AS_LIST_LOCAL "${sysbuild_SNIPPET}")
+      list(APPEND SNIPPET_AS_LIST ${SYSBUILD_SNIPPET_AS_LIST_LOCAL})
+    endif()
```

Comment:
Two comments.

First, the code in `<zephyr_base>/cmake/modules/` should be free of sysbuild specific handling.
Base code should be re-usable and sysbuild specifics should go to `<zephyr_base>/share/sysbuild/cmake/modules/` in a `sysbuild_snippets.cmake` which can then include the `snippets.cmake`.

Like we do for other CMake modules.

Second comment is for opening a discussion on which behavior we want ?
This code means that if user does `-DSNIPPET=foo -DSB_SNIPPET=bar`, then both `foo` and `bar` will be considered by `sysbuild`.

Are we sure that's really the behavior we want ?
It makes it harder for users to request sysbuild to opt-out of `foo`.

Whereas, if `SB_SNIPPET` supersede values given by `SNIPPET`, then a user wanting
- All images to use `foo` and sysbuild to use `bar`, can do: 
  `-DSNIPPET=foo -DSB_SNIPPET=bar`
- All images to use `foo` and sysbuild to use both `foo` and `bar`, can do
  `-DSNIPPET=foo -DSB_SNIPPET=foo,bar`

The latter also has the pros of allowing the user to define the order, `foo` before `bar` or `bar` before `foo`.
The current implementation has decided the order for the user.
In this case, always `foo` before `bar`.

Reply:
Reworked to support this